datadog_checks_dev/datadog_checks/dev/tooling/commands/test.py

```diff
@@ -177,6 +177,15 @@ def test(
             )
             ddtrace_check = False
 
+        # TODO: remove this when we drop PY2 support. We could also find a way to enable it only for py3 envs.
+        if any('py2' in env for env in env_names):
```

Comment:
Note: `env_names` here means "The envs that are going to be tested", not al envs. So if you run for instance `ddev test --memeray impala:py38-4.0.0`, memray will still be enabled

Reply:
However, when running on CI without specifying an environment, this will prevent memray from being used for all checks that still support py2, right?

Reply:
Yes, so basically se PR pipeline will not use memray for those integrations but the py3 only pipeline will

Reply:
Not the cleanest solution ever, but maybe we can work around this by adding a dummy `--memray` flag to our pytest plugin here when we detect python 2 (or by detecting whether the memray plugin is installed).
https://github.com/DataDog/integrations-core/blob/113d0cc41238b7074083f6c582881dab7fac4c2a/datadog_checks_dev/datadog_checks/dev/plugin/pytest.py#L317-L318

We could display a warning if the dummy flag is set (to inform the user that the option is actually being ignored).

What do you think?

Reply:
That's a cool idea. Let me test that!

Reply:
It works fine. Output for eks_fargate:

```
╰─❯ ddev test --memray eks_fargate                                                                                                                                                                   ─╯
Running tests for `eks_fargate`
-------------------------------
───────────────────────────────────────────────────────────────────────────────────────────────── py27 ─────────────────────────────────────────────────────────────────────────────────────────────────
Finished checking dependencies
cmd [1] | pytest -v --benchmark-skip tests
========================================================================================= test session starts ==========================================================================================
platform darwin -- Python 2.7.16, pytest-4.6.11, py-1.11.0, pluggy-0.13.1 -- /Users/florent.clarret/Library/Application Support/hatch/env/virtual/datadog-eks-fargate-Bbd2p0gs/py27/bin/python
cachedir: .pytest_cache
benchmark: 3.4.1 (defaults: timer=time.time disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /Users/florent.clarret/go/src/github.com/DataDog/integrations-core/eks_fargate
plugins: cov-2.12.1, benchmark-3.4.1, ddtrace-0.53.2, flaky-3.7.0, mock-2.0.0, datadog-checks-dev-17.3.2
collected 3 items                                                                                                                                                                                      

tests/test_eks_fargate.py::test_eksfargate PASSED                                                                                                                                                [ 33%]
tests/test_eks_fargate.py::test_not_eksfargate PASSED                                                                                                                                            [ 66%]
tests/test_eks_fargate.py::test_extract_resource_values PASSED                                                                                                                                   [100%]

=========================================================================================== warnings summary ===========================================================================================
/Users/florent.clarret/go/src/github.com/DataDog/integrations-core/datadog_checks_base/datadog_checks/base/utils/http.py:16
  /Users/florent.clarret/go/src/github.com/DataDog/integrations-core/datadog_checks_base/datadog_checks/base/utils/http.py:16: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.
    from cryptography.x509 import load_der_x509_certificate

/Users/florent.clarret/go/src/github.com/DataDog/integrations-core/datadog_checks_dev/datadog_checks/dev/plugin/pytest.py:336
  /Users/florent.clarret/go/src/github.com/DataDog/integrations-core/datadog_checks_dev/datadog_checks/dev/plugin/pytest.py:336: UserWarning: The --memray option can't be used with py2 environments.
    warnings.warn("The --memray option can't be used with py2 environments.")

-- Docs: https://docs.pytest.org/en/latest/warnings.html
================================================================================= 3 passed, 2 warnings in 0.51 seconds =================================================================================
───────────────────────────────────────────────────────────────────────────────────────────────── py38 ─────────────────────────────────────────────────────────────────────────────────────────────────
Finished checking dependencies
cmd [1] | pytest -v --benchmark-skip tests
========================================================================================= test session starts ==========================================================================================
platform darwin -- Python 3.8.13, pytest-7.1.3, pluggy-1.0.0 -- /Users/florent.clarret/Library/Application Support/hatch/env/virtual/datadog-eks-fargate-Bbd2p0gs/py38/bin/python
cachedir: .pytest_cache
benchmark: 4.0.0 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /Users/florent.clarret/go/src/github.com/DataDog/integrations-core/eks_fargate
plugins: ddtrace-0.53.2, flaky-3.7.0, mock-3.10.0, datadog-checks-dev-17.3.2, memray-1.3.1, cov-4.0.0, benchmark-4.0.0
collected 3 items                                                                                                                                                                                      

tests/test_eks_fargate.py::test_eksfargate PASSED                                                                                                                                                [ 33%]
tests/test_eks_fargate.py::test_not_eksfargate PASSED                                                                                                                                            [ 66%]
tests/test_eks_fargate.py::test_extract_resource_values PASSED                                                                                                                                   [100%]


============================================================================================ MEMRAY REPORT =============================================================================================
Allocations results for tests/test_eks_fargate.py::test_eksfargate

         📦 Total memory allocated: 863.2KiB
         📏 Total allocations: 1984
         📊 Histogram of allocation sizes: |█ ▂      |
         🥇 Biggest allocating functions:
                - raw_decode:/Users/florent.clarret/.pyenv/versions/3.8.13/lib/python3.8/json/decoder.py:353 -> 261.7KiB
                - __subclasscheck__:/Users/florent.clarret/.pyenv/versions/3.8.13/lib/python3.8/abc.py:102 -> 256.0KiB
                - MapItems:/Users/florent.clarret/Library/Application Support/hatch/env/virtual/datadog-eks-fargate-Bbd2p0gs/py38/lib/python3.8/site-packages/immutables/_protocols.py:44 -> 256.0KiB
                - log_typos_in_options:/Users/florent.clarret/go/src/github.com/DataDog/integrations-core/datadog_checks_base/datadog_checks/base/checks/base.py:448 -> 6.0KiB
                - __setitem__:/Users/florent.clarret/.pyenv/versions/3.8.13/lib/python3.8/os.py:681 -> 3.7KiB


Allocations results for tests/test_eks_fargate.py::test_not_eksfargate

         📦 Total memory allocated: 162.6KiB
         📏 Total allocations: 460
         📊 Histogram of allocation sizes: |▆ █ ▂   ▂|
         🥇 Biggest allocating functions:
                - decode:/Users/florent.clarret/.pyenv/versions/3.8.13/lib/python3.8/codecs.py:322 -> 73.0KiB
                - mock_from_file:/Users/florent.clarret/go/src/github.com/DataDog/integrations-core/eks_fargate/tests/test_eks_fargate.py:18 -> 73.0KiB
                - mock_from_file:/Users/florent.clarret/go/src/github.com/DataDog/integrations-core/eks_fargate/tests/test_eks_fargate.py:17 -> 4.1KiB
                - __setitem__:/Users/florent.clarret/.pyenv/versions/3.8.13/lib/python3.8/os.py:681 -> 3.7KiB
                - __setitem__:/Users/florent.clarret/.pyenv/versions/3.8.13/lib/python3.8/os.py:682 -> 2.2KiB


========================================================================================== 3 passed in 0.43s ===========================================================================================
cmd [1] | flake8 --config=../.flake8 .
cmd [2] | black --config ../pyproject.toml --check --diff .
All done! ✨ 🍰 ✨
13 files would be left unchanged.
cmd [3] | isort --settings-path ../pyproject.toml --check-only --diff .
Skipped 1 files

Passed!

```